spec/runspec.vim_spec.rb

```diff
@@ -42,15 +47,63 @@ def runspec(command)
       end
 
       it "finds a test with the same name" do
-        FileUtils.touch("test/foo_test.rb")
+        touch("test/foo_test.rb")
+        touch("foo.rb")
         expect(runspec('SpecPath("foo.rb")')).to eq("test/foo_test.rb")
       end
 
       it "finds a test with the most similar name" do
-        FileUtils.mkdir("test/unit")
-        FileUtils.touch("test/unit/user_test.rb")
+        touch("app/models/user.rb")
+        touch("test/unit/user_test.rb")
         expect(runspec('SpecPath("app/models/user.rb")')).to eq("test/unit/user_test.rb")
       end
+
+      it "finds a test from a lib directory" do
+        touch("lib/app_name/something.rb")
+        touch("test/app_name/something_test.rb")
+        expect(runspec('SpecPath("lib/app_name/something.rb")')).to eq("test/app_name/something_test.rb")
+      end
+    end
+  end
+
+  describe "#TargetPath" do
+    it "returns the current path if it doesn't contain _spec or _test" do
+      expect(runspec('TargetPath("foo.rb")')).to eq("foo.rb")
+      expect(runspec('TargetPath("bar/foo.rb")')).to eq("bar/foo.rb")
+    end
+
+    context "with a spec directory" do
+      before do
+        FileUtils.mkdir("app")
```

Comment:
Is this required or would using `touch` in the examples themselves be sufficient?